ui/src/App.js

```diff
@@ -390,6 +395,83 @@ class App extends Component {
     return filterArray;
   }
 
+  /**
+   * Converts an Array of filter strings back to a Map of filters
+   * Example:
+   * In: ["Gender=female", "Gender=male", "Population=American"]
+   * Out: {"Gender" => ["male", "female"], "Population" => ["American"]}
+   */
+  filterArrayToMap(filterArray) {
+    let filterMap = new Map();
+    filterArray.forEach(function(pair) {
+      pair = pair.split("=");
+      if (filterMap.has(pair[0])) {
+        let arr = filterMap.get(pair[0]);
+        arr.push(pair[1]);
+        filterMap.set(pair[0], arr);
+      } else {
+        filterMap.set(pair[0], [pair[1]]);
+      }
+    });
+    return filterMap;
+  }
+
+  handleQueryString() {
+    let queryStringJSON = this.queryStringToJSON();
+    this.callFacetsAPIGet(
+      {
+        filter: queryStringJSON.filter,
+        extraFacets: queryStringJSON.extraFacets
+      },
+      function(error, data) {
+        this.facetsCallback(error, data);
+        this.setState({
+          // Set selectedFacetValues state after facetsCallback.
+          // If it were set before, the relevant facet might not yet be in extraFacetEsFieldsNames.
+          selectedFacetValues: this.filterArrayToMap(queryStringJSON.filter),
+          extraFacetEsFieldNames: queryStringJSON.extraFacets
+        });
+      }.bind(this)
+    );
+  }
+
+  /**
+   * Converts the query string parameters in the current window
+   * to a dictionary object usable by the backend
+   * Example:
+   * In: http://data-explorer.com/?filter=Gender%3Dfemale%7CGender%3Dmale%7CPopulation%3DAmerican&extraFacets=Population
+   * Out: {"filter" => ["Gender=female", "Gender=male", "Population=American"], "extraFacets" => ["Population"]}
+   */
+  queryStringToJSON() {
+    var pairs = window.location.search.slice(1).split("&");
+    var result = {};
+    pairs.forEach(function(pair) {
+      pair = pair.split("=");
+      result[pair[0]] = pair[1] ? decodeURIComponent(pair[1]).split("|") : [];
+    });
+    return result;
+  }
+
+  updateQueryParams(facetFilterParam, extraFacetsParam) {
+    window.history.pushState(
+      null,
+      "",
+      "?filter=" + facetFilterParam + "&extraFacets=" + extraFacetsParam
+    );
+  }
+
+  // Wrap the call to facetsGet together with updating the query params
+  callFacetsAPIGet(data, facetsCallback) {
```

Comment:
nit callFacetsApiGet

That's what style guides for most languages do (eg [js](https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined)).

Also queryStringToJson